crates/gitbutler-branch-actions/src/move_commits.rs

```diff
@@ -70,32 +72,50 @@ pub(crate) fn move_commit(
         .find_commit(merge_base)
         .context("failed to find merge base")?;
 
-    let branch_head_diff: HashMap<_, _> =
-        gitbutler_diff::diff_files_into_hunks(branch_head_diff).collect();
-    let is_source_locked = check_source_lock(source_branch_non_comitted_files, &branch_head_diff);
+    let source_commit_diff: HashMap<_, _> =
+        gitbutler_diff::diff_files_into_hunks(source_commit_diff).collect();
+    let is_source_locked = check_source_lock(source_branch_non_comitted_files, &source_commit_diff);
 
-    let mut ancestor_commits = ctx.repository().log(
-        source_branch_head_parent.id(),
-        LogUntil::Commit(merge_base.id()),
-    )?;
+    let mut ancestor_commits = ctx
+        .repository()
+        .log(source_commit_parent.id(), LogUntil::Commit(merge_base.id()))?;
     ancestor_commits.push(merge_base);
-
     let ancestor_commits = ancestor_commits;
 
+    let mut descendant_commits = None;
+    if !is_head_commit {
+        descendant_commits = Some(
+            ctx.repository()
+                .log(source_branch.head(), LogUntil::Commit(commit_id))?,
+        );
+    }
+
     let is_ancestor_locked =
-        check_source_lock_to_ancestors(ctx.repository(), ancestor_commits, &branch_head_diff);
+        check_source_lock_to_commits(ctx.repository(), &ancestor_commits, &source_commit_diff);
 
     if is_source_locked {
         bail!("the source branch contains hunks locked to the target commit")
     }
 
     if is_ancestor_locked {
-        bail!("the source branch contains hunks locked to the target commit ancestors")
+        bail!("the target commit contains hunks locked to its ancestors")
+    }
+
+    if let Some(commits_to_check) = descendant_commits.as_mut() {
+        // we append the source commit so that we can create the diff between
+        // the source commit and its first descendant
+        commits_to_check.push(source_commit.clone());
```

Comment:
This is a little fix I did on top.
We need to append the source commit to the commit vector we pass into the `check_source_lock_to_commits` in because that's how the diffs are calculated in that function.

That's not immediately clear nor explained anywhere, so that's my bad!

The `check_source_lock_to_commits` function takes in the diff of the commit we're attempting to move and its parent.
It then calculates all of the diffs of the passed in commits.

So if we have a branch with commits `A, B, C, D` (`A` being the top-most commit) and want to move `C`, to check if it's locked to it's descendants (`A` and `B`), we need to pass in the diff between `C` and `D` (`source_commit_diff`) and the commits `A, B, C` (`commits`).

The function will calculate, in order, the diff between `A` - `B` and `B` - `C`. And compare each diff to see if it overlaps with the `source_commit_diff` lines